apps/genpkey.c

```diff
@@ -196,9 +196,9 @@ int genpkey_main(int argc, char **argv)
 
     if (text) {
         if (do_param)
-            rv = EVP_PKEY_print_params(out, pkey, 0, NULL);
+            rv = EVP_PKEY_print_params(bio_out, pkey, 0, NULL);
```

Comment:
Use bio_out for the text - it seems silly to output it to the same place as the pem/der file.

Reply:
Not sure when it comes to PEM. Have you never seen PEM files that include a text printout of the object? I have seen a lot of certs made that way. This is perhaps not quite as valuable with keys, but...
Either way, the PEM readers are designed to handle random text surrounding the PEM payload.

Reply:
So just revert it you think?

Reply:
Yes